lib/config.js

```diff
@@ -20,14 +22,51 @@ function Config () {
 	for (var key in config) {
 		this[key] = config[key];
 	}
+	
+	if (mode === 'worker'){
+		if (!this.output_root_dir) {
+			this.output_root_dir = '~/Guerilla';
+		}
+		this.output_root_dir = this.expandTilde(this.output_root_dir);
+		mkdirp.sync(this.output_root_dir);
+	}
+
 }
 
+
 function getUrlObj (host, port) {
 	if (host.slice(-1) === '/') host = host.slice(0, -1);
 	var urlObj = url.parse(host + ':' + port + '/');
 	return urlObj;
 }
 
+Config.prototype.getMode = function () {
+	if (argv.master) return 'master';
+	else if (argv.worker) return 'worker';
+	else throw new Error('Invalid mode.');
+}
+
+
+Config.prototype.expandTilde = function (apath) {
+	var homedir = process.env['HOME'];
```

Comment:
getMode is arguably part of the config abstraction. expandTilde is util*ish, but we cannot easily support circular dependencies. So we wither move expandTIlde to a util2 which doesn't depend on anything, or we leave this alone.